tests/ui/lint/anonymous-reexport.rs

```diff
@@ -0,0 +1,16 @@
+#![deny(useless_anonymous_reexport)]
+#![crate_type = "rlib"]
+
+mod my_mod {
+    pub trait Foo {}
+    pub type TyFoo = dyn Foo;
+    pub struct Bar;
+    pub type TyBar = Bar;
+}
+
+pub use self::my_mod::Foo as _;
+pub use self::my_mod::TyFoo as _;
+pub use self::my_mod::Bar as _; //~ ERROR
+pub use self::my_mod::TyBar as _; //~ ERROR
```

Comment:
Should we suggest to remove it?
Could you add tests with nested uses:
```rust
pub use self::my_mod::{Bar as _};
pub use self::my_mod::{Bar as _, Foo as _};
pub use self::my_mod::{Bar as _, TyBar as _};
```

Reply:
Adding tests with nested uses.

Reply:
> Should we suggest to remove it?

I'm not sure. If you prefer I add this suggestion, I can.